CesiumGS/cesium
# Add TS types for `Check`, remove circular dependency

# Description

Adds types for the `Check` group of functions with type assertions to help in TS based projects.
I chose to add this as a `Check.d.ts` file right next to `Check.js` to keep them closer together. `tsd-jsdoc` doesn't seem to have a way to import this types file though so it still needs a custom replace in our generate TS build step.

Also removed a circular dependency I accidentally added in #11859 







## Issue number and link

No issue





## Testing plan

- Run `npm run build-ts` and make sure it succeeds
- Check `Source/Ceisum.d.ts` and `packages/engine/index.d.ts` that they have the correct definition for `Check`
- Run `npm run build-docs` and make sure it succeeds
- Check that the Documentation now has `Check` on the globals page.

For the circular dependency you can use [`dpdm`](https://github.com/acrazing/dpdm) to check in `main` and this branch with the command `npx dpdm packages/engine/Source/DataSources/CzmlDataSource.js`



# Author checklist
